source/adapters/opencl/context.cpp

```diff
@@ -169,6 +170,13 @@ UR_APIEXPORT ur_result_t UR_APICALL urContextCreateWithNativeHandle(
 UR_APIEXPORT ur_result_t UR_APICALL urContextSetExtendedDeleter(
     ur_context_handle_t hContext, ur_context_extended_deleter_t pfnDeleter,
     void *pUserData) {
+  if (!ur::cl::getAdapter()->clSetContextDestructorCallback) {
```

Comment:
NIT: I feel `ur::cl::getAdapter()` could be saved in a variable but that is just styling thing so feel free to ignore.

Reply:
I will disagree with you on this point.